src/Channels/MemoryEnumerator.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Channels
+{
+    /// <summary>
+    /// An enumerator over the <see cref="ReadableBuffer"/>
+    /// </summary>
+    public struct MemoryEnumerator
+    {
+        private ReadableBuffer _buffer;
+        private Memory<byte> _current;
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="buffer"></param>
+        public MemoryEnumerator(ref ReadableBuffer buffer)
+        {
+            _buffer = buffer;
+            _current = default(Memory<byte>);
+        }
+
+        /// <summary>
+        /// The current <see cref="Span{Byte}"/>
+        /// </summary>
+        public Memory<byte> Current => _current;
+
+        /// <summary>
+        /// 
+        /// </summary>
+        public void Dispose()
```

Comment:
That shows whether its called or not; isn't any code in Dispose, so behaviour is the same. Though I wonder if it can just be taken out...